scripts/train/train.py

```diff
@@ -32,95 +32,131 @@
                                        build_composer_model, build_evaluators,
                                        build_logger, build_optimizer,
                                        build_scheduler, build_tokenizer)
-from llmfoundry.utils.config_utils import (log_config, pop_config,
-                                           process_init_device,
+from llmfoundry.utils.config_utils import (TRAIN_CONFIG_KEYS, TrainConfig,
+                                           log_config,
+                                           make_dataclass_and_log_config,
+                                           pop_config, process_init_device,
+                                           to_dict_container, to_list_container,
                                            update_batch_size_info)
 from llmfoundry.utils.registry_utils import import_file
 
 log = logging.getLogger(__name__)
 
 
-def validate_config(cfg: DictConfig):
+def validate_config(train_config: TrainConfig):
     """Validates compatible model and dataloader selection."""
-    loaders = [cfg.train_loader]
-    if 'eval_loader' in cfg:
-        eval_loader = cfg.eval_loader
-        if isinstance(eval_loader, ListConfig):
-            for loader in eval_loader:
-                if loader.label is None:
-                    raise ValueError(
-                        'When specifying multiple evaluation datasets, each one must include the \
+    # Check for missing mandatory fields
+    for field in TRAIN_CONFIG_KEYS:
```

Comment:
does the call to structured not handle this already?

Reply:
The call to structured doesn't error on missing mandatory keys at initialization time, it errors at access time